### ISSUE TITLE
[Xenon] SEMI_AUTOMATIC mode blocks loop() after calling Mesh.off()

### ISSUE THREAD

**Issue by ScruffR** [particle-iot/device-os#1813]
After the first publish the code never runs `loop()` again.

```
SYSTEM_MODE(SEMI_AUTOMATIC);

void setup() {
  Serial.begin();
}

void loop() {
  static uint32_t msLast = 0;
  Serial.print('.');             // <-- get only printed on first call to loop() !!!
  if (millis() - msLast < 1000) return;
  msLast = millis();

  Serial.println("Turning ON radio...");
  Mesh.on();
  Mesh.connect(); 
  if (waitFor(Mesh.ready, 10000)) {
    Particle.connect();
    if (waitFor(Particle.connected, 10000)) {
      Particle.publish("xenon-test", PRIVATE);  // tried with WITH_ACK too 
      Serial.println("published xenon-text");   //  |
      delay(100);                               // delay required (even when using WITH_ACK) 
    }
    else
      Serial.println("Couldn't connect to cloud");
  }
  else
    Serial.println("Couldn't connect to mesh");
  
  Serial.println("Turning OFF radio...");
  Mesh.off();
  Serial.println("radio is off");
}
```

This works in `AUTOMATIC` and `MANUAL` mode tho'
When also using `SYSTEM_THREAD(ENABLED)` in `SEMI_AUTOMATIC` mode `loop()` keeps getting called but the device throws a single red blink (no SOS panic) after a few iterations.

Tested with 1.1.0 and 1.2.1-rc.2 (Xenon + Argon 1.1.0 gateway)

**Comment by YutingYou** [comment]
Hi @ScruffR, thanks for reporting this issue, it has been fixed in #1857 .

**Comment by avtolstoy** [comment]
Closing as PR #1857 has been merged.
